### PR TITLE
Add request response fields to webhook service

### PR DESCRIPTION
**Pull request by buildbreaker**
Webhook rpc methods:
- CreateWebhook: creates a webhook for a repository
- DeleteWebhook: delets a webhook for a repository
- ListWebhooks: lists webhooks for a repository

Currently there is only one webhook event type: repository push (`WEBHOOK_EVENT_REPOSITORY_PUSH`)

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/webhook.proto [hunk 1]
```diff
@@ -17,26 +17,82 @@ syntax = "proto3";
...
+message CreateWebhookRequest {
+  // The events to subscribe to for the given repository.
+  repeated WebhookEvent event = 1;
+  // The owner name of the repository in the corresponding subscription request.
+  string owner_name = 2;
+  // The repository name that the subscriber wishes create a subscription for.
+  string repository_name = 3;
+  // The subscriber's callback URL where notifications
+  // should be delivered. It is considered good practice
+  // to use a unique callback URL for each subscription.
```
Comment (rubensf) [inline proto/buf/alpha/registry/v1alpha1/webhook.proto:41]: `It is considered good practice
  // to use a unique callback URL for each subscription.`

I'm generally of the opinion that usage guidelines like this shouldn't live in the protobuf definitions - particularly without more context (why?). But I'm not sure what are our current practices about those, and I don't feel strongly about it.

File: proto/buf/alpha/registry/v1alpha1/webhook.proto [hunk 2]
```diff
@@ -17,26 +17,82 @@ syntax = "proto3";
...
+}
 
-// CreateRepositoryResponse is the proto repsonse representation
-// of a webhook request to subscribe.
-message SubscribeToRepositoryResponse {}
+// CreateWebhookResponse is the proto repsonse representation
+// of a webhook request.
+message CreateWebhookResponse {
+  // For the user to unsubscribe from the webhook event.
+  string webhook_id = 1;
```
Comment (rubensf) [inline proto/buf/alpha/registry/v1alpha1/webhook.proto:58]: Call it `webhook_subscription_id` for consistency?

**Review by rubensf — APPROVED** [review]
One comment & one nit. Don't forget to sync merging this without a counterpart internal PR

File: proto/buf/alpha/registry/v1alpha1/webhook.proto [hunk 3]
```diff
@@ -17,26 +17,80 @@ syntax = "proto3";
...
 }
 
-// SubscribeToRepositoryRequest is the proto request representation of a
-// webhook request body to subscribe.
-message SubscribeToRepositoryRequest {}
+// CreateWebhookRequest is the proto request representation of a
+// webhook request body.
+message CreateWebhookRequest {
+  // The events to subscribe to for the given repository.
+  repeated WebhookEvent event = 1;
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/webhook.proto:34]: `webhook_event`

File: proto/buf/alpha/registry/v1alpha1/webhook.proto [hunk 4]
```diff
@@ -17,26 +17,80 @@ syntax = "proto3";
...
+  repeated SubscribedWebhook subscribed_webhooks = 1;
+  // The next page token for paginating.
+  string next_page_token = 2;
+}
+
+// SubscribedWebhook is the representation of a webhook event and webhook
+// subscription id.
+message SubscribedWebhook {
+  // The event associated with the subscription id.
+  WebhookEvent event = 1;
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/webhook.proto:93]: `webhook_event`

**Review by bufdev — APPROVED** [review]
Two field name changes, otherwise LGTM!

**Pull request merged**